Add Wellcome OR latex template

arnold-c opened the pull request:
To contribute a new article template to this package, please make sure you have done the following things (note that `journalname_article` below is only an example name):

- [x] This project uses a Contributor Licence Agreement (CLA) that you'll be asked to sign when opening a PR. This is required for a significant pull request (it is fine not to sign it if a PR is only intended to fix a few typos). We use a tool called CLA assistant for that.  
You could also, unless you have done it in any other RStudio's projects before, sign the [individual](https://rstudioblog.files.wordpress.com/2017/05/rstudio_individual_contributor_agreement.pdf) or [corporate](https://rstudioblog.files.wordpress.com/2017/05/rstudio_corporate_contributor_agreement.pdf) contributor agreement. You can send the signed copy to <jj@rstudio.com>.

- [x] Add the `journalname_article()` function to `R/article.R` if the output format is simple enough, otherwise create a separate `R/journalname_article.R`.

- [x] Add the Pandoc LaTeX template `inst/rmarkdown/templates/journalname/resources/template.tex`.

- [x] Add a skeleton article `inst/rmarkdown/templates/journalname/skeleton/skeleton.Rmd`.

- [x] Add a description of the template `inst/rmarkdown/templates/journalname/template.yaml`.

- [x] Please include the document class file (`*.cls`) if needed, but please do not include standard LaTeX packages (`*.sty`) that can be downloaded from CTAN. If you are using TinyTeX or TeX Live, you can verify if a package is available on CTAN via `tinytex::parse_packages(files = "FILENAME"")` (e.g., when `FILENAME` is `plain.bst`, it should return `"bibtex"`, which means this file is from a standard CTAN package). Please keep the number of new files absolutely minimal (e.g., do not include PDF output files), and also make examples minimal (e.g., if you need a `.bib` example, try to only leave one or two bibliography entries in it, and don't include too many items in it without using all of them).

- [x] Update Rd and namespace (could be done by `devtools::document()`).

- [x] Update NEWS.

- [x] Update README with a link to the newly supported journal. Please add your Github username and the full name of the journal (follow other examples in the list). 

- [x] Add a test to `tests/testit/test-formats.R` by adding a line `test_format("journalname")`. We try to keep them in alphabetical order.

- [x] Add your name to the list of authors `Authors@R` in DESCRIPTION. You don't need to bump the package version in DESCRIPTION.

Lastly, please try your best to do only one thing per pull request (e.g., if you want to add two output formats, do them in two separate pull requests), and refrain from making cosmetic changes in the code base: https://yihui.name/en/2018/02/bite-sized-pull-requests/

Thank you!

---------------------------------

Hi,

I'm afraid I'm relatively inexperienced at both package development and LaTeX, so have adapted the template as much as I can, but am currently running into a problem when I try to knit `skeleton.Rmd`: `Error: 'wellcomeor_article' is not an exported object from 'namespace:rticles'`. If anyone can point me in the right direction, that'd be much appreciated, and I can see if the template works as expected.

Thanks,

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rstudio/rticles?pullRequest=436) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rstudio/rticles?pullRequest=436) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rstudio/rticles?pullRequest=436) it.</sub>

cderv commented:
Thanks for this addition ! 

The error you get is because you did not roxygenize the package to transform the `@export` to a line in the NAMESPACE. I'll push the change so that you can continue without this error

cderv commented:
Thanks for the contribution of this new template. 

> but am currently running into a problem when I try to knit skeleton.Rmd: Error: 'wellcomeor_article' is not an exported object from 'namespace:rticles'

Did you install the development version of this package (from your branch) ? 

Using the Knit button will use the install version. If you are using `devtools::load_all()` you would need to use command line `rmarkdown::render` I think

arnold-c commented:
Sorry for the delay in response @cderv, and thanks for reminding me to build and install the development version! I'm now getting this issue

```

processing file: skeleton.Rmd
  |......................................................................| 100%
   inline R code fragments


/Applications/RStudio.app/Contents/MacOS/pandoc/pandoc +RTS -K512m -RTS skeleton.knit.md --to latex --from markdown+autolink_bare_uris+tex_math_single_backslash --output skeleton.tex --lua-filter /Users/cfa5228/Library/R/x86_64/4.1/library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /Users/cfa5228/Library/R/x86_64/4.1/library/rmarkdown/rmarkdown/lua/latex-div.lua --self-contained --template /Users/cfa5228/Library/R/x86_64/4.1/library/rticles/rmarkdown/templates/wellcomeor/resources/template.tex --highlight-style tango --pdf-engine pdflatex --citeproc 
output file: skeleton.knit.md

Error compiling template "/Users/cfa5228/Library/R/x86_64/4.1/library/rticles/rmarkdown/templates/wellcomeor/resources/template.tex" (line 77, column 1):
unexpected end of input
expecting " ", "\t", "$--", "\n", "\r\n", "\r", "$", "${" or "$$"
Error: pandoc document conversion failed with error 5
Execution halted
```

Looking in both `skeleton.Rmd` and `template.tex` I can't see any code that would be causing these issues, but I'm obviously missing something. I've tried removing the majority of the `body` text in an attempt to see if some of the LaTeX code was causing the error, but this didn't seem to help.

cderv commented:
This type of error is thrown by Pandoc when something is not correct regarding syntax in the template. There must be a Pandoc template syntax not correct in the template. It could be an `$endif$` missing or any other `$` missing around variables.

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/skeleton.Rmd:
```diff
@@ -0,0 +1,154 @@
+
+date: "`r Sys.Date()`"
+bibliography: sample.bib
+csl: elsevier-vancouver.csl
```
Could we suggest here to use the URL for this CLS, and add a comment to indicate to download and use local if one prefer ? This would be a way for us to avoid including the CSL file and having a potential outdated one in the package

```suggestion
# You can download the csl file if preferred and use it locally with `csl: elsevier-vancouver.csl`
csl: https://www.zotero.org/styles/elsevier-vancouver
```

cderv reviewed (changes requested):
I have left a few comments. 

The CI issue is because a `$endfor$` is missing I believe

Also you have some file in this PR the concerns the `arxiv` template, in the arxiv folder. `skeleton.pdf` and `skeleton.tex` ? Why are they here ? 
I think they should be removed. 

I did not do a thorough review yet but can you also fill the template with some missing pieces required by Pandoc ? 
At least 
https://github.com/rstudio/rticles/blob/2e0a3d02b197aa1b6e888a8ffa9e34fdf1c925e5/inst/rmarkdown/templates/acm/resources/template.tex#L60-L97

After theses changes I'll test further the template and check if everything is there.

arnold-c commented:
Thanks @cderv. I've added the Pandoc requirements you linked, along with your other suggestions, and removed the arXiv files that accidentally were made when I was examining the template.

It is no longer throwing the Pandoc error, but I am now getting the following error

```
! Undefined control sequence.
<argument> ...sed on Wellcome Open Research \href 
                                                  {https://www.overleaf.com/...
l.62 Latex template}}
```

`l.62` refers to 

```
\emph{Text based on Wellcome Open Research
\href{https://www.overleaf.com/project/6131911644b635ad3aaa4cb2}{Overleaf
Latex template}}
```

When removing `_Text based on Wellcome Open Research [Overleaf Latex template](https://www.overleaf.com/project/6131911644b635ad3aaa4cb2)_` from `skeleton.Rmd`, I instead get the following error:

```
! Undefined control sequence.
l.60 \hypertarget
                 {introduction}{% 
```

Which refers to: 
```
\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}
```

Thanks

arnold-c commented:
I've updated the template and it now works. The only remaining issues I couldn't figure out is how to disable link coloring (setting it using `hypersetup{}` didn't seem to work), and the author affiliation number isn't correct and doesn't allow for multiple authors. I tried to copy across code from the OUP template, which allows for it, but couldn't get it to work I'm afraid. I also get font warning, but it seems like that is also an warning on the [Overleaf template](https://www.overleaf.com/project/6131911644b635ad3aaa4cb2), so I don't think it's too much of an issue that should be fixed on our end.

```
Warning message:
LaTeX Warning: Font shape declaration has incorrect series value `mc'.
               It should not contain an `m'! Please correct it.
               Found on input line 53.
```

cderv commented:
> I couldn't figure out is how to disable link coloring

It is possible this is something produced by Pandoc. Did you try opting out using the variables ? 
https://pandoc.org/MANUAL.html#links

Something like 
````yaml
colorlinks: false
````
in the YAML header ?

> the author affiliation number isn't correct and doesn't allow for multiple authors. 

This can be tricky and require to modify the template. Several formats have different trick for this kind of thing. 
I could try to have a look. That would be helpful if you know what syntax it should look like in the `.tex` output ? 
We could then adapt as template for Pandoc. 

Thanks

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/skeleton.Rmd:
```diff
@@ -0,0 +1,196 @@
+bibliography: sample.bib
+biblio-style: "unsrtnat"
+link-citations: TRUE
+colorlinks: TRUE
```
This is what creates your link coloring. You should not set it if you don't want link color.

arnold-c replied:
I was having issues with this not doing what I expected, but I think that's been addressed in a recent commit

cderv replied:
There could be another way - to follow pandoc template directly. 
https://github.com/jgm/pandoc/blob/master/data/templates/default.latex

I'll propose you some changes toward that.

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/skeleton.Rmd:
```diff
@@ -0,0 +1,196 @@
+colorlinks: TRUE
+# You can download the csl file if preferred and use it locally with `csl: elsevier-vancouver.csl`
+csl: https://www.zotero.org/styles/elsevier-vancouver
+output: 
+  bookdown::pdf_book:
+    base_format: rticles::wellcomeor_article
+    citation_package: natbib
+    number_sections: FALSE
```
We can show this also, but it would be best to show only article also

```suggestion
output: 
  rticles::wellcomeor_article:
    citation_package: natbib
  bookdown::pdf_book:
    base_format: rticles::wellcomeor_article
    citation_package: natbib
    number_sections: FALSE
```

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/skeleton.Rmd:
```diff
@@ -0,0 +1,196 @@
+colorlinks: TRUE
+# You can download the csl file if preferred and use it locally with `csl: elsevier-vancouver.csl`
+csl: https://www.zotero.org/styles/elsevier-vancouver
+output: 
+  bookdown::pdf_book:
+    base_format: rticles::wellcomeor_article
+    citation_package: natbib
+    number_sections: FALSE
```
Does this need to be set all the time ? If so we can change the default in the format function.

arnold-c replied:
Are you referring to the `number_sections` line? If so, I've set that to default to false in https://github.com/rstudio/rticles/pull/436/commits/4bb1beb4a38f526a4d0cf64db4949f73bbe50b40, so hopefully that resolves this concern, but please let me know if not

cderv replied:
It was not directly a concern. 
My question was: "Does the template requires `number_sections` to be FALSE by default ? "

If so, then we should just to it in the format function
```diff
diff --git a/R/article.R b/R/article.R
index 1f09012..ad9aa96 100644
--- a/R/article.R
+++ b/R/article.R
@@ -446,11 +446,11 @@ trb_article <- function(..., keep_tex = TRUE, citation_package = 'natbib') {
 }
 
 #' @section \code{wellcomeor_article}: Format for creating submissions to
 #' Wellcome Open Research. Adapted from \url{https://www.overleaf.com/project/6131911644b635ad3aaa4cb2}.
 #' @export
 #' @rdname article
-wellcomeor_article <- function(..., keep_tex = TRUE, citation_package = 'natbib') {
+wellcomeor_article <- function(..., number_sections = FALSE, keep_tex = TRUE, citation_package = 'natbib') {
   pdf_document_format(
-    "wellcomeor", keep_tex = keep_tex, citation_package = citation_package, ...
+    "wellcomeor", keep_tex = keep_tex, number_sections = number_sections, citation_package = citation_package, ...
   )
 }
```

and we would leave the template as the own in Pandoc
https://github.com/jgm/pandoc/blob/9cbea695c439dd04c9a670107ec729c503d5b368/data/templates/default.latex#L318-L322

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/skeleton.Rmd:
```diff
@@ -0,0 +1,196 @@
+
+Please do not list grant funding in this section.
+
+# References
+
+References can be listed in any standard referencing style that uses a numbering
+system(i.e. not Harvard referencing style), and should be consistent between
+references within a given article.
+
+Reference management systems such as Zotero provide options for exporting
+bibliographies as Bib\TeX{} files. Bib\TeX{} is a bibliographic tool that is
+used with \LaTeX{} to help organize the user's references and create a
+bibliography. This template contains an example of such a file,
+\texttt{sample.bib}, which can be replaced with your own. You can \LaTeX{}
+command  \verb|\cite| to create in-text citationsor by using
+RMarkdown/Bookdown's citations [@Smith:2012qr], and [@Smith:2013jd]
```
By using a section References in the example there will be two sections references in the PDF Output

Probably need to change the title here ? 
![image](https://user-images.githubusercontent.com/6791940/143575423-6836ef61-aa62-4199-8895-a5cb559648f3.png)

arnold-c replied:
Changed the title of this section in https://github.com/rstudio/rticles/pull/436/commits/d2a84aee3a42ed3ca43085011f61702c1603f0eb

cderv commented on inst/rmarkdown/templates/wellcomeor/template.yaml:
```diff
@@ -0,0 +1,6 @@
+name: Wellcome Open Research Journal Article
+description: >
+ Template for creating an article for submission to Wellcome Open Research
+
```
```suggestion
```

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,143 @@
+% \usepackage[round]{natbib}
+% \let\cite\citep
+
+% Pandoc toggle for numbering sections (defaults to be off)
+$if(numbersections)$
+$else$
+\setcounter{secnumdepth}{0}
+$endif$
```
Not sure to understand why we need this 🤔 

Can you explain from your tries ? 

Sorry if I already asked, but I don't remember.

arnold-c replied:
Because I'm not too familiar with latex and pandoc templates, I just copied some code that seemed to do what I wanted and remove section numbering. I think I've cleaned it up in https://github.com/rstudio/rticles/pull/436/commits/4bb1beb4a38f526a4d0cf64db4949f73bbe50b40 to set that as a default without the unnecessary `if` command

cderv replied:
Sections numbers are supposed to be off by default ? Wasn't it the case in the default ? 

It would be better to stick with Pandoc's template and set `number_sections` to FALSE in the function, just in case the default change

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,143 @@
+$endfor$
+
+\begin{document}
+
```
We just need to add a part for Pandoc feature here: 

Example:
https://github.com/rstudio/rticles/blob/3b2e6a7499058b48b0de684b28925b14bb544eed/inst/rmarkdown/templates/ams/resources/template.tex#L109-L112

arnold-c replied:
I have added the `include-before` lines before the `$body$` call in https://github.com/rstudio/rticles/pull/436/commits/faa2f5cc99d4ec4cddf6cf4aa6d9588416fa1ee5. Is this OK?

cderv replied:
Yes thanks

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,143 @@
+
+% We hope you find the Wellcome Open Research Overleaf template useful,
+% please let us know if you have any feedback using the help menu above.
+
```
We also need a Pandoc part here just in case 
Example
https://github.com/rstudio/rticles/blob/3b2e6a7499058b48b0de684b28925b14bb544eed/inst/rmarkdown/templates/ams/resources/template.tex#L114-L117

arnold-c replied:
Done in https://github.com/rstudio/rticles/pull/436/commits/36fdf8a823f3d0ec0ecb18d4e0e82684331671bb

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/elsevier-vancouver.csl:
```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" encoding="utf-8"?>
```
I don't think we need this file as the skeleton is showing example with a URL. 

We should delete it.

cderv reviewed (changes requested):
Hey ! 

What is the status on this PR ? 

I reviewed it as is after merging changes from the main branch to be up to date.

Overall it looks good ! Small questions and feedback before we can merge. 

Thanks !

cderv commented on NEWS.md:
```diff
@@ -4,7 +4,9 @@
-- Add `trb_article()` for annual meeting submissions to the Transportation Research Board Annual Meeting (thanks, @gregmacfarlane, #427).
+- New `trb_article()` for annual meeting submissions to the Transportation Research Board Annual Meeting (thanks, @gregmacfarlane, #427).
+
+- New `wellcomeor_article()` for Wellcome Open Research articles
```
```suggestion
- New `wellcomeor_article()` for Wellcome Open Research articles (thanks, @arnold-c, #436).
```

cderv commented on R/article.R:
```diff
@@ -444,3 +444,13 @@ trb_article <- function(..., keep_tex = TRUE, citation_package = 'natbib') {
 }
+
+#' @section \code{wellcomeor_article}: Format for creating submissions to
+#' Wellcome Open Research. Adapted from \url{https://www.overleaf.com/project/6131911644b635ad3aaa4cb2}.
```
I did not checked previously but this URL does not work. It is restricted. 
Is it this one ? https://www.overleaf.com/latex/templates/wellcome-open-research-article-template/hsmhhbpxvvbj

```suggestion
#' Wellcome Open Research. Adapted from <https://www.overleaf.com/project/6131911644b635ad3aaa4cb2>.
```

arnold-c replied:
Yes, thanks for catching this

cderv commented on R/article.R:
```diff
@@ -444,3 +444,13 @@ trb_article <- function(..., keep_tex = TRUE, citation_package = 'natbib') {
   )
 }
+
+#' @section \code{wellcomeor_article}: Format for creating submissions to
```
Just that we can use markdown in roxygen comment now https://roxygen2.r-lib.org/articles/rd-formatting.html#links-2

```suggestion
#' @section `wellcomeor_article`: Format for creating submissions to
```

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,152 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% For more detailed article preparation guidelines, please see: http://wellcomeopenresearch.org/for-authors/article-guidelines and http://wellcomeopenresearch.org/for-authors/data-guidelines
+
```
This goes with the proposal change on hyperlinks color
```suggestion
% Options for packages loaded elsewhere - from Pandoc's template
\PassOptionsToPackage{unicode$for(hyperrefoptions)$,$hyperrefoptions$$endfor$}{hyperref}
\PassOptionsToPackage{hyphens}{url}
$if(colorlinks)$
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
$endif$
```

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,152 @@
+% \usepackage[round]{natbib}
+% \let\cite\citep
+
+% Pandoc toggle for numbering sections (defaults to be off)
+\setcounter{secnumdepth}{0}
```
What if we stick to Pandoc's template ? 
```r
number_section = FALSE
```

should not make section numbered

```suggestion
% Pandoc toggle for numbering sections
$if(numbersections)$
\setcounter{secnumdepth}{$if(secnumdepth)$$secnumdepth$$else$5$endif$}
$else$
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
$endif$
```

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,152 @@
+\providecommand{\tightlist}{%
+  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
+
+%% Default: numerical citations
+\usepackage[numbers]{natbib}
+
+%% Uncomment this lines for superscript citations instead
+% \usepackage[super]{natbib}
+
+%% Uncomment these lines for author-year citations instead
+% \usepackage[round]{natbib}
+% \let\cite\citep
```
Let's just add the flag in case someone change `citation_package`. And also add the option so that the other option can be used

```suggestion
$if(natbib)$
\usepackage[$if(natbiboptions)$$natbiboptions$$else$numbers$endif$]{natbib}
$endif$
```

That way you can set 

````yaml
natbiboptions: super # for superscript citations instead
````
or 
````yaml
natbiboptions: round # for author-year citations instead
````

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,152 @@
+% hyperref required
+\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
+
+% Link coloring
+\hypersetup{$if(colorlinks)$
+            ,
+            colorlinks=$colorlinks$,
+            citecolor=$if(citecolor)$$citecolor$$else$blue$endif$,
+            urlcolor=$if(urlcolor)$$urlcolor$$else$blue$endif$,
+            linkcolor=$if(linkcolor)$$linkcolor$$else$blue$endif$
+            $else$
+            hidelinks,
+            $endif$
+}
```
We can use the same as in Pandoc template here to get the full options (https://github.com/jgm/pandoc/blob/9cbea695c439dd04c9a670107ec729c503d5b368/data/templates/default.latex#L215)
```suggestion
% Link coloring
\hypersetup{
$if(title-meta)$
  pdftitle={$title-meta$},
$endif$
$if(author-meta)$
  pdfauthor={$author-meta$},
$endif$
$if(lang)$
  pdflang={$lang$},
$endif$
$if(subject)$
  pdfsubject={$subject$},
$endif$
$if(keywords)$
  pdfkeywords={$for(keywords)$$keywords$$sep$, $endfor$},
$endif$
$if(colorlinks)$
  colorlinks=true,
  linkcolor={$if(linkcolor)$$linkcolor$$else$Maroon$endif$},
  filecolor={$if(filecolor)$$filecolor$$else$Maroon$endif$},
  citecolor={$if(citecolor)$$citecolor$$else$Blue$endif$},
  urlcolor={$if(urlcolor)$$urlcolor$$else$Blue$endif$},
$else$
  hidelinks,
$endif$
  pdfcreator={LaTeX via pandoc}}
```

If `colorlinks` is not set in YAML (the default), then links should not be colored and `hidelinks` will be used instead. 
That is how `colorlinks` works

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/skeleton.Rmd:
```diff
@@ -0,0 +1,204 @@
+bibliography: sample.bib
+biblio-style: "unsrtnat"
+link-citations: TRUE
+colorlinks: FALSE # See hypersetup in template.tex for link color options
```
```suggestion
```

I think we don't need that as it is off by default

cderv commented on inst/rmarkdown/templates/wellcomeor/skeleton/skeleton.Rmd:
```diff
@@ -0,0 +1,204 @@
+biblio-style: "unsrtnat"
+link-citations: TRUE
+colorlinks: FALSE # See hypersetup in template.tex for link color options
+# You can download the csl file if preferred and use it locally with `csl: elsevier-vancouver.csl`
+csl: https://www.zotero.org/styles/elsevier-vancouver
```
CSL is useful only when Pandoc citeprocessing is used. Your default is `natbib` so let's include but comment

```suggestion
# Setting CSL will be necessary if you use `citation_packahe = "default"`
# You can download the csl file if preferred and use it locally with `csl: elsevier-vancouver.csl`
# csl: https://www.zotero.org/styles/elsevier-vancouver
```

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,152 @@
+$endfor$
+* Corresponding author\\
+
+$for(include-before)$
+  $include-before$
+$endfor$
```
Just need a space
```suggestion
$for(include-before)$
  $include-before$

$endfor$
```

cderv commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,152 @@
+% We hope you find the Wellcome Open Research Overleaf template useful,
+% please let us know if you have any feedback using the help menu above.
+
+$for(include-after)$
+$include-after$
+$endfor$
```
and here also

```suggestion
$for(include-after)$
$include-after$

$endfor$
```

cderv reviewed (changes requested):
Hi, 

thanks a lot for the feedback. We are preparing a releasing so this is good news if we can merge this. 

Based on your feedback, and my previous work this past two weeks on **rticles** I did more comment. Instead of just making them and pushing, I made them as suggestion so that you can validate if that is ok to you. 

My aim is for the template to be closest to Pandoc's one so that features work and it is easy to update. 

Please, have a look (below and above) and confirm or not  my sugestion. Once I'll have your ok, I'll commit and merge everything. 

Tests are now passing - some contents where missing.

arnold-c commented on inst/rmarkdown/templates/wellcomeor/resources/template.tex:
```diff
@@ -0,0 +1,152 @@
+\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
+
+% Link coloring
+\hypersetup{$if(colorlinks)$
+            ,
+            colorlinks=$colorlinks$,
+            citecolor=$if(citecolor)$$citecolor$$else$blue$endif$,
+            urlcolor=$if(urlcolor)$$urlcolor$$else$blue$endif$,
+            linkcolor=$if(linkcolor)$$linkcolor$$else$blue$endif$
+            $else$
+            hidelinks,
+            $endif$
+}
```
Looks good - I was having an issue where if I didn't specify `colorlinks` in the YAML, but also didn't have `hidelinks` in the `template.tex`, it would draw an outline around the citation links. Is this the expected behavior? Either way, the suggested change looks good to me.

arnold-c commented:
All look good to me so I just went ahead and committed the suggestions. Thanks

arnold-c commented:
Sorry about that. I've just gone through and confirmed it all works as expected after those suggestions (I think I got them all)

cderv reviewed (approved):
Awesome that is great ! 

We are ready to roll then. Thank you for this  PR and you're availability to make the necessary changes!

cderv commented:
We must just redocument following new arguments in function. I'll do that tommorow and merge it.
